backend/tournesol/views/inconsistencies.py

```diff
@@ -0,0 +1,213 @@
+    """
+
+    serializer_class = ScoreInconsistenciesSerializer
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        poll = self.poll_from_url
+        user = request.user
+
+        filter_serializer = ScoreInconsistenciesFilterSerializer(data=request.query_params)
+        filter_serializer.is_valid(raise_exception=True)
+        filters = filter_serializer.validated_data
+
+        contributor_comparisons_criteria = ComparisonCriteriaScore.objects.filter(
+            comparison__user=user,
+            comparison__poll=poll,
+        )
+
+        if filters["date_gte"]:
+            contributor_comparisons_criteria = contributor_comparisons_criteria.filter(
+                comparison__datetime_lastedit__gte=filters["date_gte"]
+            )
+        contributor_comparisons_criteria = list(
+            contributor_comparisons_criteria.values(
+                "comparison__entity_1__uid",
+                "comparison__entity_2__uid",
+                "criteria",
+                "score",
+            )
+        )
+
+        ratings = list(
+            ContributorRatingCriteriaScore.objects.filter(
+                contributor_rating__user=user,
+                contributor_rating__poll=poll,
+            ).values(
+                "contributor_rating__entity__uid",
+                "criteria",
+                "uncertainty",
+                "score",
+            )
+        )
+
+        response = self._list_inconsistent_comparisons(contributor_comparisons_criteria,
+                                                       ratings,
+                                                       filters["inconsistency_threshold"])
+
+        return Response(ScoreInconsistenciesSerializer(response).data)
+
+    @staticmethod
+    def _list_inconsistent_comparisons(criteria_comparisons: list,
+                                       criteria_ratings: list,
+                                       threshold: float) -> dict:
+        """
+        For each comparison criterion, search the corresponding rating,
+        calculate the inconsistency, and check if it crosses the threshold.
+        Then prepare the HTTP response.
+        """
+
+        count_comparisons_analysed = 0
+        inconsistency_sum = 0.0
+        inconsistent_criterion_comparisons = []
+
+        for comparison_criterion in criteria_comparisons:
+
+            entity_1 = comparison_criterion["comparison__entity_1__uid"]
+            entity_2 = comparison_criterion["comparison__entity_2__uid"]
+            criterion = comparison_criterion["criteria"]
+            comparison_score = comparison_criterion["score"]
+
+            entity_1_score = None
+            entity_2_score = None
+            entity_1_uncertainty = None
+            entity_2_uncertainty = None
+            # TODO: after passing to Python 3.10, sort criteria_ratings and optimize with bisect
```

Comment:
I have let a TODO.  It would be much more efficient to sort the criteria_ratings list first and then use dichotomy, but bisect only offers the "key" parameter since python 3.10 (https://docs.python.org/3/library/bisect.html).

Reply:
This "key" parameter would allow to use dichotomy on both "criteria" and "contributor_rating__entity__uid" values of the dictionary (without the "key" parameter, bisect is not suited for dictionaries).

Reply:
yes, it's a better idea